File: Exercises2024/Chapter05_basic_md/C/2-Ewald/Source/realspace.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "system.h"

// real Part; also direct calculation 
// loop over all particle pairs
void Realspace(double *Ureal)
{
  int i,j;
  VECTOR dr;
  double U,r,r2;

  U=0.0;

  // start modification
  // 1. For all particle pairs calculate the distance in x, y, and z.
  // 2. Apply periodic boundary conditions where necessary.
  // 3. Calculate the real-space contribution to the energy. 
  // (Use the "ErrorFunctionComplement(x)" function.) 


  // end  modification
  *Ureal=U;
}

File: Exercises2024/Chapter05_basic_md/Fortran/2-Ewald/Source/realspace.f
      Subroutine Realspace(Ureal)
      Implicit None

      Include 'system.inc'

Cccccccccccccccccccccccccccccccccccccccccccc
C     Real Path; Also Direct Calculation   C
C     Loop Over All Particle Pairs         C
Cccccccccccccccccccccccccccccccccccccccccccc

      Integer I,J
      Double Precision Dx,Dy,Dz,Ureal,R,R2,Ir,Dderfc

      Ureal = 0.0d0

C     Start Modification
c    1. For all particle pairs calculate the distance in x, y, and z.
c    2. Apply periodic boundary conditions where necessary.
c    3. Calculate the real-space contribution to the energy. 
c    (Use the "Dderfc(x)" function to calculate the Error-Function Complement.) 

C     End   Modification

      Return
      End
